hazelcast-hibernate5/src/main/java/com/hazelcast/hibernate/telemetry/PhoneHomeService.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2020 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package com.hazelcast.hibernate.telemetry;
+
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
+
+import java.io.BufferedInputStream;
+import java.io.InputStream;
+import java.net.URL;
+import java.net.URLConnection;
+import java.time.Duration;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static com.hazelcast.internal.nio.IOUtil.closeResource;
+import static java.lang.Boolean.FALSE;
+import static java.lang.System.getenv;
+import static java.util.concurrent.Executors.newSingleThreadScheduledExecutor;
+
+/**
+ * Pings phone home server with plugin information daily.
+ *
+ * @since 2.1.2
+ */
+public class PhoneHomeService {
+
+    private static final String SYS_PHONE_HOME_ENABLED = "hazelcast.phone.home.enabled";
+    private static final String ENV_PHONE_HOME_ENABLED = "HZ_PHONE_HOME_ENABLED";
+
+    private static final String PHONE_HOME_URL = "http://phonehome.hazelcast.com/pingIntegrations/hazelcast-hibernate5";
+    private static final Duration TIMEOUT = Duration.ofMillis(3000);
+
+    private final ILogger logger = Logger.getLogger(PhoneHomeService.class);
+    private final AtomicBoolean started = new AtomicBoolean();
+    private final ScheduledExecutorService executor = newSingleThreadScheduledExecutor(r
+            -> new Thread(r, "PhoneHomeService"));
+
+    private PhoneHomeInfo phoneHomeInfo;
+
+    public void start() {
+        if (started.compareAndSet(false, true) && isPhoneHomeEnabled()) {
+            executor.scheduleAtFixedRate(this::send, 0, 1, TimeUnit.DAYS);
+        }
+    }
+
+    private static boolean isPhoneHomeEnabled() {
+        String falseStr = FALSE.toString();
+        if (falseStr.equalsIgnoreCase(getenv(ENV_PHONE_HOME_ENABLED))) {
+            return false;
+        }
+        if (falseStr.equalsIgnoreCase(System.getProperty(SYS_PHONE_HOME_ENABLED))) {
+            return false;
+        }
```

Comment:
What should we do if someone sets one to _false_ and the other to _true_? :) how do other modules handle it?

Reply:
MC does not check env and core checks the inverse order. 
Changing the order of env and sys here to make it core alike:
https://github.com/hazelcast/hazelcast/blob/36d11eb5c1cfeeb11be3a0e53136fbcb2767f6da/hazelcast/src/main/java/com/hazelcast/internal/util/PhoneHome.java#L80